### PR TITLE
Add K2SO implementation

### PR DESCRIPTION
**Pull request by Frugghi**
image

### CODE REVIEW

File: test/server/cards/01_SOR/units/K2SOCassiansCounterpart.spec.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+                context.player1.clickCard(context.k2so);
+                context.player1.clickCard(context.kraytDragon);
+                expect(context.p2Base.damage).toBe(2);
+
+                expect(context.player1).toHaveExactPromptButtons(['Deal 3 damage to opponent\'s base', 'The opponent discards a card']);
+                expect(context.player1).not.toHavePassAbilityButton();
+
+                context.player1.clickPrompt('Deal 3 damage to opponent\'s base');
+                expect(context.p2Base.damage).toBe(5);
```
Comment (VJubert) [inline test/server/cards/01_SOR/units/K2SOCassiansCounterpart.spec.ts:49]: I think you should add a test with no hand for opponent to see if they is not prompt to choose between both choice

Comment (Frugghi) [reply test/server/cards/01_SOR/units/K2SOCassiansCounterpart.spec.ts:49]: Lines 29-36 are covering the case where the opponent doesn't have any card. I'm not sure the option should be auto selected though, wouldn't be possible to pick "discard a card" even if the opponent has no card? I don't think there is a requirement to change the game state here.

File: server/game/cards/01_SOR/units/K2SOCassiansCounterpart.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+export default class K2SOCassiansCounterpart extends NonLeaderUnitCard {
+    protected override getImplementationId() {
+        return {
+            id: '3232845719',
+            internalName: 'k2so#cassians-counterpart',
+        };
+    }
+
+    public override setupCardAbilities() {
+        // TODO: This doesn't support Twin Sun correctly
```
Comment (AMMayberry1) [inline server/game/cards/01_SOR/units/K2SOCassiansCounterpart.ts:14]: Fine to leave this here, but for the future - there are so many places where this is true that we definitely shouldn't try to mark all of them

File: test/server/cards/01_SOR/units/K2SOCassiansCounterpart.spec.ts (hunk 1 above)
Comment (AMMayberry1) [reply test/server/cards/01_SOR/units/K2SOCassiansCounterpart.spec.ts:49]: Yeah the option should still be available

**Review by AMMayberry1 — APPROVED** [review]

**Pull request merged**